src/metaopt/io/parsing.py

```diff
@@ -0,0 +1,257 @@
+
+    """
+    defcfg = nesteddict()
+
+    # get some defaults
+    defcfg['exp_name'] = DEF_CMD_EXP_NAME[1].format(
+        user=user,
+        starttime=starttime.isoformat())
+    defcfg['max_trials'] = DEF_CMD_MAX_TRIALS[0]
+    defcfg['pool_size'] = DEF_CMD_POOL_SIZE[0]
+
+    # get default options for some managerial variables (see `ENV_VARS`)
+    for signif, evars in six.iteritems(ENV_VARS):
+        for _, key, default_value in evars:
+            defcfg[signif][key] = default_value
+
+    for cfgpath in DEF_CONFIG_FILES_PATHS:
+        try:
+            with open(cfgpath) as f:
+                cfg = yaml.safe_load(f)
+                if cfg is None:
+                    continue
+                # implies that yaml must be in dict form
+                for k, v in six.iteritems(cfg):
+                    if k in ENV_VARS:
+                        for vk, vv in six.iteritems(v):
+                            defcfg[k][vk] = vv
+                    else:
+                        defcfg[k] = v
+        except IOError as e:  # default file could not be found
+            log.debug(e)
+        except AttributeError as e:
+            log.warn("Problem parsing file: %s", cfgpath)
+            log.warn(e)
+
+    return defcfg
+
+
+def env_vars(config):
+    """
+    Fetches environmental variables related to metaopt's managerial data.
+
+    :type config: `nesteddict`
+
+    """
+    newcfg = deepcopy(config)
+    for signif, evars in six.iteritems(ENV_VARS):
+        for var_name, key, default_value in evars:
+            value = os.getenv(var_name)
+            if value is not None:
+                newcfg[signif][key] = value
+    return newcfg
+
+
+def mopt_config(config, dbconfig, cmdconfig, cmdargs):
+    """
+    'moptconfig' can describe:
+       * 'name': Experiment's name. If you provide a past experiment's name,
+         then that experiment will be resumed. This means that its history of
+         trials will be reused, along with any configurations logged in the
+         database which are not overwritten by current call to `mopt` script.
+         (default: <username>_<start datetime>)
+       * 'max_trials': Maximum number of trial evaluations to be computed
+         (required as a cmd line argument or a moptconfig parameter)
+       * 'pool_size': Number of workers evaluating in parallel asychronously
+         (default: 10 @ default resource). Can be a dict of the form:
+         {resource_alias: subpool_size}
+       * 'database': (db_opts)
+       * 'resources': {resource_alias: (entry_address, scheduler, scheduler_ops)}
+         (optional)
+       * 'optimizer': {optimizer module name : method-specific configuration}
+       * 'dynamic': {dynamic module name : method-specific configuration}
+
+       .. seealso:: Method-specific configurations reside in `/config`
+
+    """
+    expconfig = deepcopy(config)
+
+    for cfg in (dbconfig, cmdconfig):
+        for k, v in six.iteritems(cfg):
```

Comment:
Isn't there a similar loop inside `default_options`?

Reply:
Indeed

Reply:
That loop is over 'configuration paths'. I think it's an overkill to abstract it to a function.

Reply:
Fine. I just wanted to make sure there is no avoidable duplications.